NsqSharp/Consumer.cs

```diff
@@ -1383,6 +1391,81 @@ private void redistributeRDY()
             }
         }
 
+        private void redistributeRDYForStarvedConnections(List<Conn> connections, int maxInFlight)
+        {
+            var activeConns = new List<Conn>();
+            var idleConns = new List<Conn>();
+
+            foreach (var c in connections)
+            {
+                var lastMsgDuration = DateTime.Now.Subtract(c.LastMessageTime);
+                long rdyCount = c.RDY;
+                if (rdyCount > 0 && lastMsgDuration > _config.LowRdyIdleTimeout)
+                {
+                    idleConns.Add(c);
+                    _needRdyRedistributed = 1;
+                }
+                else if (lastMsgDuration <= _config.LowRdyIdleTimeout ||
+                         (rdyCount == 0 && lastMsgDuration > _config.LowRdyIdleTimeout))
+                {
+                    activeConns.Add(c);
+                    if (rdyCount == 0)
+                        _needRdyRedistributed = 1;
+                }
+            }
+
+            if (Interlocked.CompareExchange(ref _needRdyRedistributed, value: 0, comparand: 1) != 1)
+            {
+                return;
+            }
+
+            long availableMaxInFlight = maxInFlight - _totalRdyCount;
```

Comment:
@judwhite I think `availableMaxInFlight` captures the value of `_totalRdyCount` here, but that state might change on line 1449 before it's used again on line 1452.